runtime/compiler/net/ServerStream.hpp

```diff
@@ -73,17 +74,26 @@ class ServerStream : public CommunicationStream
    virtual ~ServerStream()
       {
       _numConnectionsClosed++;
+      _pClientSessionData = NULL;
       }
 
    /**
       @brief Send a message to the client
 
       @param [in] type Message type to be sent
-      @param [in] args Variable number of additional paramaters to be sent
+      @param [in] args Variable number of additional parameters to be sent
    */
    template <typename ...Args>
    void write(MessageType type, Args... args)
       {
+      // TODO: Depending on where the mutex is taken for read,
+      // we probably don't need to check the message CHTable_getClassInfoUpdates.
+      if (isReadingClassUnload() && isClassUnloadingAttempted())
```

Comment:
Is the condition on `isReadingClassUnload()` needed?
Is this the filter for catching the case when we need to send one message while doing class unloading?

Reply:
`isReadingClassUnload()` is added to avoid aborting the compilation due to the messages sent in `processUnloadedClasses()` such as `MessageType::getUnloadedClassRanges`. Inside `processUnloadedClasses()`, `isClassUnloadingAttempted()` returns true. And also if more messages are added in the future when `isClassUnloadingAttempted()` is true, `isReadingClassUnload()` should take care of it instead of filtering by specific message types.